### ISSUE TITLE
add Channel pixel order

### ISSUE THREAD

**Issue by roipoussiere** [Ge0rg3/StegOnline#30]
Let's assume that we want to encode 24 bits of data in a 2x2 image (using a fictive 4 bits per channel image format, for readability). The `X` represents places where bits will be altered.

For now, 2 options are available in the Pixel Ordel list:

## Column

```
  r   g   b      r   g   b
╔═══╦═══╦═══╗  ╔═══╦═══╦═══╗
║ X ║ X ║ X ║  ║   ║   ║   ║
╠═══╬═══╬═══╣  ╠═══╬═══╬═══╣
║ X ║ X ║ X ║  ║   ║   ║   ║
╠═══╬═══╬═══╣  ╠═══╬═══╬═══╣
║ X ║ X ║ X ║  ║   ║   ║   ║
╠═══╬═══╬═══╣  ╠═══╬═══╬═══╣
║ X ║ X ║ X ║  ║   ║   ║   ║
╚═══╩═══╩═══╝  ╚═══╩═══╩═══╝
╔═══╦═══╦═══╗  ╔═══╦═══╦═══╗
║ X ║ X ║ X ║  ║   ║   ║   ║
╠═══╬═══╬═══╣  ╠═══╬═══╬═══╣
║ X ║ X ║ X ║  ║   ║   ║   ║
╠═══╬═══╬═══╣  ╠═══╬═══╬═══╣
║ X ║ X ║ X ║  ║   ║   ║   ║
╠═══╬═══╬═══╣  ╠═══╬═══╬═══╣
║ X ║ X ║ X ║  ║   ║   ║   ║
╚═══╩═══╩═══╝  ╚═══╩═══╩═══╝
```

Result (200x163 image, 73kb of data):
![image](https://user-images.githubusercontent.com/1665542/146531377-6475f263-53a0-4a06-990b-b371b4e14ab8.png)

## Row

```
  r   g   b      r   g   b
╔═══╦═══╦═══╗  ╔═══╦═══╦═══╗
║ X ║ X ║ X ║  ║ X ║ X ║ X ║
╠═══╬═══╬═══╣  ╠═══╬═══╬═══╣
║ X ║ X ║ X ║  ║ X ║ X ║ X ║
╠═══╬═══╬═══╣  ╠═══╬═══╬═══╣
║ X ║ X ║ X ║  ║ X ║ X ║ X ║
╠═══╬═══╬═══╣  ╠═══╬═══╬═══╣
║ X ║ X ║ X ║  ║ X ║ X ║ X ║
╚═══╩═══╩═══╝  ╚═══╩═══╩═══╝
╔═══╦═══╦═══╗  ╔═══╦═══╦═══╗
║   ║   ║   ║  ║   ║   ║   ║
╠═══╬═══╬═══╣  ╠═══╬═══╬═══╣
║   ║   ║   ║  ║   ║   ║   ║
╠═══╬═══╬═══╣  ╠═══╬═══╬═══╣
║   ║   ║   ║  ║   ║   ║   ║
╠═══╬═══╬═══╣  ╠═══╬═══╬═══╣
║   ║   ║   ║  ║   ║   ║   ║
╚═══╩═══╩═══╝  ╚═══╩═══╩═══╝
```

Result:
![image](https://user-images.githubusercontent.com/1665542/146531349-be47e1ec-239c-4bce-b27d-d54f66ba8762.png)

## Channel

And here is my proposal: iterating over channels instead rows or columns:

```
  r   g   b      r   g   b
╔═══╦═══╦═══╗  ╔═══╦═══╦═══╗
║ X ║ X ║ X ║  ║ X ║ X ║ X ║
╠═══╬═══╬═══╣  ╠═══╬═══╬═══╣
║ X ║ X ║ X ║  ║ X ║ X ║ X ║
╠═══╬═══╬═══╣  ╠═══╬═══╬═══╣
║   ║   ║   ║  ║   ║   ║   ║
╠═══╬═══╬═══╣  ╠═══╬═══╬═══╣
║   ║   ║   ║  ║   ║   ║   ║
╚═══╩═══╩═══╝  ╚═══╩═══╩═══╝
╔═══╦═══╦═══╗  ╔═══╦═══╦═══╗
║ X ║ X ║ X ║  ║ X ║ X ║ X ║
╠═══╬═══╬═══╣  ╠═══╬═══╬═══╣
║ X ║ X ║ X ║  ║ X ║ X ║ X ║
╠═══╬═══╬═══╣  ╠═══╬═══╬═══╣
║   ║   ║   ║  ║   ║   ║   ║
╠═══╬═══╬═══╣  ╠═══╬═══╬═══╣
║   ║   ║   ║  ║   ║   ║   ║
╚═══╩═══╩═══╝  ╚═══╩═══╩═══╝
```

Result:
![image](https://user-images.githubusercontent.com/1665542/146531075-348bf8c3-e548-48b1-9df0-700c0d873b92.png)

This is an interesting feature as it avoids to visually see the difference between clean an encoded parts, and improve global image quality.

**Comment by Ge0rg3** [comment]
Hi! Thank you for your suggestion. This would definitely be an interesting feature, and I would love to include this in the next release. However, it seems a bit tricky from a UI standpoint -- any ideas on this?

**Comment by Ge0rg3** [comment]
Hi, on further thought, isn't this already implemented via the following config?
![image](https://user-images.githubusercontent.com/16269580/156886991-38d51c2f-f9d2-42ac-9818-a06a7f1908a0.png)

**Comment by roipoussiere** [comment]
No, this produces an image with altered pixels on top of the image. What I suggest is to distribute altered pixels on the whole image.

On my test scenario, 7 on 8 bits must be altered in order to encode the whole data in the image (this way we can clearly see the difference in the image).

So, with pixel order = row, bit order = MSB, bit plane order = rgb, pad remaining bits = no, and these options:
⬜ ⬜ ⬜
☑️ ☑️ ☑️
☑️ ☑️ ☑️
☑️ ☑️ ☑️
☑️ ☑️ ☑️
☑️ ☑️ ☑️
☑️ ☑️ ☑️
☑️ ☑️ ☑️
produce this:
![image](https://user-images.githubusercontent.com/1665542/159705492-d723d56a-3d02-4ab6-b95e-1a21b6632129.png)

And these options:
☑️ ☑️ ☑️
☑️ ☑️ ☑️
☑️ ☑️ ☑️
☑️ ☑️ ☑️
☑️ ☑️ ☑️
☑️ ☑️ ☑️
☑️ ☑️ ☑️
⬜ ⬜ ⬜
produce this:
![image](https://user-images.githubusercontent.com/1665542/159705638-d5d871f5-788d-4549-96a1-18e5b67ec9ee.png)

Here are my test data:

input: *[Le Désespéré](https://en.wikipedia.org/wiki/Le_D%C3%A9sesp%C3%A9r%C3%A9), Gustave Courbet*
![Le_Désespéré](https://user-images.githubusercontent.com/1665542/159706102-9efa18f1-5cf4-42d2-a222-c945cead9532.png)


image file to encode: *[The Starry Night](https://en.wikipedia.org/wiki/The_Starry_Night), Vincent van Gogh*
![The Starry Night](https://user-images.githubusercontent.com/1665542/159706388-79bfc428-0516-4080-93a1-7ecd89ebded2.png)
